plugin/src/main/scala/com/olegpy/bm4/NoMapIdentity.scala

```diff
@@ -45,11 +45,28 @@ trait NoMapIdentity extends TreeUtils {
           mods.flags == ModifierFlags.PARAM &&
           (i.tpe =:= tpt.tpe) =>
         true
-      case Function(ValDef(_, _, tpt, EmptyTree) :: Nil, u @ Literal(Constant(())))
+      case Function(ValDef(_, _, tpt, EmptyTree) :: Nil, LiteralUnit(u))
         if tpt.tpe =:= definitions.UnitTpe && tpt.tpe =:= u.tpe =>
         true
       case _ =>
         false
     }
   }
+
+  object LiteralUnit {
+    def unapply(tree: Tree): Option[Tree] = tree match {
+      case u @ Literal(Constant(())) =>
+        Some(u)
+
+      // In Scala 2:13, we get `(x$1: Unit @unchecked) match { case _ => () }`
+      case u @ Match(Typed(Ident(TermName(x)), tpt),
```

Comment:
I couldn't figure out how to target this with a quasiquote.  I _think_ this won't allow anything but a simple type ascripted Unit as the scrutinee, rather than a side effect, but I don't know what I'm doing.

Reply:
Yes, it should work this way. The tree at place `Literal(Constant(()))` would be a `Block` if there was any funny business.

Reply:
I tried to clean it up with a quasiquote something like `q"($expr: $type) match { case _ => () }`, but whenever I progressed to `q($expr: Unit @unchecked) match { case _ => () }` it ceased to match.  I couldn't figure out how to validate that `$type` is `Unit` without going back to the tree extractors.

Reply:
I remember there was some funny business because `@` can be an annotation and a pattern bind (e.g. in `case (a @ List(_))`) and I was only able to [resolve it using tree extractors too](https://github.com/oleg-py/better-monadic-for/blob/master/src/main/scala/com/olegpy/bm4/NoUncheckedFilter.scala#L20), so that is perfectly fine as a solution. My comment was regarding side-effects.